pandas/core/arrays/base.py

```diff
@@ -431,6 +431,11 @@ def astype(self, dtype, copy=True):
         array : ndarray
             NumPy ndarray with 'dtype' for its dtype.
         """
+        from pandas.core.arrays.string_ import StringDtype
```

Comment:
so we atcually *ever* hit this base type? I think we override this everywhere

Reply:
floats arrays are PandasArrays, and those get their astype from ExtensionsArray.

```python
>>> pd.array([1.5, 2.5])
<PandasArray>
[1.5, 2.5]
Length: 2, dtype: float64
```

Allowing any ExtensionArray by default to convert to StringArray seems reasonable to me (and if subclassers don't want that, they can make their own astype implementation disallowing StringArrays).

Reply:
kk, can you add a commment to this effect here (agree with your statement), but comment for future readers

Reply:
> can you add a commment to this effect here

All methods in this base class are there for subclasses to (potentially) use, so I don't think a comment about that is needed 
(a comment about always being able to astype to string dtype is fine though)

Reply:
i understand I would move the impl that is currently in Decimal to here as it correctly handles the astype from the *same* type (whereas this one will coerce to a numpy array)